lib/core/src/display/shape/text/text_field.rs

```diff
@@ -185,19 +185,51 @@ shared! { TextField
 }
 
 
+// === Constructor ===
+
+impl TextField {
+    /// Create new TextField.
+    pub fn new(world:&World, initial_content:&str, properties:TextFieldProperties) -> Self {
+        let data              = TextFieldData::new(world,initial_content,properties);
+        let rc                = Rc::new(RefCell::new(data));
+        let frp               = TextFieldFrp::new(Rc::downgrade(&rc));
+        with(rc.borrow_mut(), move |mut data| {
+            data.keyboard_binding = Some(frp.bind_frp_to_js_text_input_actions());
+            data.frp              = Some(frp);
+        });
+        Self{rc}
+    }
+}
+
+
 // === Private ===
 
 impl TextFieldData {
-    fn initialize(&mut self, fonts:&mut FontRegistry) {
-        self.assignment_update(fonts).update_line_assignment();
-        self.rendered.update_glyphs(&mut self.content,fonts);
-        self.rendered.update_cursor_sprites(&self.cursors, &mut self.content.full_info(fonts));
+    fn new(world:&World, initial_content:&str, properties:TextFieldProperties) -> Self {
+        let logger           = Logger::new("TextField");
+        let display_object   = DisplayObjectData::new(logger);
+        let content          = TextFieldContent::new(initial_content,&properties);
+        let cursors          = Cursors::default();
+        let rendered         = TextFieldSprites::new(world,&properties);
+        let frp              = None;
+        let keyboard_binding = None;
+        display_object.add_child(rendered.display_object.clone_ref());
+
+        let mut text_field = Self
+            {properties,content,cursors,rendered,display_object,frp,keyboard_binding};
+        text_field.initialize();
+        text_field
```

Comment:
Lets make `initialize` taking ownership of self and simplify these lines

Reply:
I will simplify here, but desimplify in `initialize()` function, so what is the point?